### DISCUSSION TITLE
Problem with displaying SmartDataframes in Streamlit

### DISCUSSION THREAD

**Discussion by notmanas** [sinaptik-ai/pandas-ai#506] [Q&A]
I have been using pandasai version 0.8.1 by calling PandasAI API. But that will be deprecated soon, so I switched to SmartDataframe.
Earlier if I got a dataframe back from PandasAI after processing, I could just use -

`st.dataframe(df, use_container_width=True)`

to show that dataframe. Now, it doesn't work that way anymore. If I use st.write(result_df), I can see it as a table -
![st write output](https://github.com/gventuri/pandas-ai/assets/59655018/2c4416a9-b779-4953-8019-4fe358ad358f)

BUT if I use 
`st.dataframe(result_df)`

I don't see the column headers anymore? -
![st df output](https://github.com/gventuri/pandas-ai/assets/59655018/5fe34511-9f56-48ac-bc4a-cb8063d4a538)

How do I tackle this?

Thanks in advance!

**Comment by fabmeyer** [comment]
Having the same problem, please help!

**Reply by fabmeyer** [reply]
A quick fix:

```
smart_dataframe = SmartDataframe(dataframe, config={'llm': llm})
answer = smart_dataframe.chat(user_input)
column_names = answer.columns
answer = pd.DataFrame(data=answer, columns=column_names)
```

**Comment by gventuri** [comment]
@notmanas @fabmeyer this shouldn't happen. The logic is pretty much the same.
Maybe one attempt could be:

```
smart_dataframe = SmartDataframe(dataframe, config={'llm': llm})
df = smart_dataframe.chat(user_input)
answer = df.dataframe # this will return the original pandas dataframe instead of a smart dataframe
```

Let me know if it works

**Reply by fabmeyer** [reply]
Same what I did:
https://github.com/gventuri/pandas-ai/discussions/506#discussioncomment-7457136
